### ISSUE TITLE
bug: false positive for no-pass-ref-to-parent

### ISSUE THREAD

**Issue by nwalters512** [NickvanDyke/eslint-plugin-react-you-might-not-need-an-effect#58]
### Did you check docs and existing issues?

- [x] I have read the plugin docs
- [x] I have searched the existing issues
- [x] I have read [You Might Not Need An Effect](https://react.dev/learn/you-might-not-need-an-effect) and am reasonably confident that I *do* need an effect

### Plugin version

0.8.3

### ESLint version

9.39.2

### Describe the bug

I'm receiving false positives from the following code:

```js
import React, { useEffect, useRef } from "react";

// This is from TanStack Virtualizer in the real codebase.
function useVirtualizer({ getScrollElement }) {
  return {
    measure: () => {
      console.log("Measuring columns...", getScrollElement());
    },
  };
}

export function MyComponent() {
  const scrollContainerRef = useRef(null);

  const columnVirtualizer = useVirtualizer({
    getScrollElement: () => scrollContainerRef.current,
  });

  useEffect(() => {
    columnVirtualizer.measure();
  }, [columnVirtualizer]);

  return <div ref={scrollContainerRef}>Hello, world!</div>;
}
```

Specifically, the `react-you-might-not-need-an-effect/no-pass-ref-to-parent` rule flags three separate issues with the `columnVirtualizer.measure()` line.

### Steps To Reproduce

1. Clone the repo from https://github.com/nwalters512/no-pass-ref-to-parent-repro
2. Follow the instructions in the `README.md`
3. Observe the following warnings:

```
  20:5  warning  Avoid passing refs to parents in an effect. Use `forwardRef` instead                                                                   react-you-might-not-need-an-effect/no-pass-ref-to-parent
  20:5  warning  Avoid calling props inside callbacks registered on refs in an effect. Use `forwardRef` to register the callback in the parent instead  react-you-might-not-need-an-effect/no-pass-ref-to-parent
  20:5  warning  Avoid receiving refs from parents to use in an effect. Use `forwardRef` instead                                                        react-you-might-not-need-an-effect/no-pass-ref-to-parent
```

If you want to see the real code in context, you can check out https://github.com/PrairieLearn/PrairieLearn/pull/13646. The reproduction I shared above is as minimal as I could get it.

### Expected Behavior

I would not expect to see any warnings from this lint rule for this code, as it's not passing refs to a parent or receiving any refs from a parent.

It seems like the lint rule is somehow picking up on the fact that `columnVirtualizer` comes from `useVirtualizer`, and `useVirtualizer` uses a ref as part of `getScrollElement`? AFAICT this is totally unrelated to "passing refs to parents" - there's not even any parent involved here!
